packages/client/example.js

```diff
@@ -4,64 +4,40 @@
 
 process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
 
-const {client, xml} = require('./index') // For you; require('@xmpp/client')
+const {xml, client} = require('./index') // For you; require('@xmpp/client')
 const entity = client()
 
-// Emitted for any error
+// Log errors
 entity.on('error', err => {
-  console.error('error', err)
+  console.error('❌', err.toString())
 })
 
-entity.on('close', () => {
-  console.log('closed')
+// Log status changes
+entity.on('status', (status, value) => {
+  console.log('🛈', status, value ? value.toString() : '')
 })
 
-// Emitted for incoming stanza _only_ (iq/presence/message) qualified with the right namespace
-// entity.on('stanza', (stanza) => {
-//   console.log('stanza', stanza.toString())
-// })
-
-// Emitted for incoming nonza _only_
-// entity.on('nonza', (nonza) => {
-//   console.log('nonza', nonza.toString())
-// })
-
-// useful for logging raw traffic
+// Useful for logging raw traffic
 // Emitted for every incoming fragment
-entity.on('input', data => console.log('⮈ IN ', data))
+entity.on('input', data => console.log('⮈', data))
 // Emitted for every outgoing fragment
-entity.on('output', data => console.log('⮊ OUT', data))
-
-// Emitted for any in our out XML root element
-// useful for logging
-// entity.on('element', (input, output) => {
-//   console.log(output ? 'element =>' : 'element <=', (output || input).toString())
-// })
-
-// Emitted when the connection is established
-entity.on('connect', () => {
-  console.log('1. connected')
-})
-
-// Emitted when the XMPP stream has open and we received the server stream
-entity.on('open', () => {
-  console.log('2. open')
+entity.on('output', data => console.log('⮊', data))
+
+// Useful for logging XML traffic
+// Emitted for every incoming XML element
+// entity.on('element', data => console.log('⮈', data))
+// Emitted for every outgoing XML element
+// entity.on('send', data => console.log('⮊', data))
+
+entity.on('stanza', el => {
+  if (el.is('presence') && el.attrs.from === entity.jid.toString()) {
+    console.log('🗸', 'available, ready to receive <message/>')
+  }
 })
 
-// Emitted when the XMPP entity is authenticated
-entity.on('authenticated', () => {
-  console.log('3. authenticated')
-})
-
-// Emitted when authenticated and bound
 entity.on('online', jid => {
```

Comment:
authenticated happens before online

(online for client is usually after resource binding)